### PR TITLE
Add long error explanation for E0588

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #61137.

r? @kinnison

### CODE REVIEW

File: src/librustc_typeck/error_codes.rs [hunk 1]
```diff
@@ -3891,6 +3891,33 @@ details.
...
+```compile_fail,E0588
+#[repr(align(16))]
+struct Aligned(i32);
+
+#[repr(packed)] // error!
+struct Packed(Aligned);
+```
+
+Just like you can't have both `align` and `packed` representation hints on a
+same type, a type can't contain another type with the `align` representation
```
Comment (kinnison) [inline src/librustc_typeck/error_codes.rs]:
```suggestion
same type, a `packed` type can't contain another type with the `align` representation
```

**Comment by GuillaumeGomez** [comment]
Updated.

**Review by kinnison — APPROVED** [review]
LGTM

File: src/librustc_typeck/error_codes.rs [hunk 2]
```diff
@@ -3891,6 +3891,33 @@ details.
...
+
+```compile_fail,E0588
+#[repr(align(16))]
+struct Aligned(i32);
+
+#[repr(packed)] // error!
+struct Packed(Aligned);
+```
+
+Just like you can't have both `align` and `packed` representation hints on a
```
Comment (Centril) [inline src/librustc_typeck/error_codes.rs]:
```suggestion
Just like you cannot have both `align` and `packed` representation hints on a
```

File: src/librustc_typeck/error_codes.rs [hunk 3]
```diff
@@ -3891,6 +3891,33 @@ details.
...
+same type, a `packed` type can't contain another type with the `align`
```
Comment (Centril) [inline src/librustc_typeck/error_codes.rs]:
```suggestion
same type, a `packed` type cannot contain another type with the `align`
```

**Comment by kinnison** [comment]
@Dylan-DPC :+1:

**Pull request merged**